browser/default_protocol_handler_utils_win.cc

```diff
@@ -0,0 +1,505 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/default_protocol_handler_utils_win.h"
+
+#include <bcrypt.h>
+#include <shobjidl.h>
+#include <wincrypt.h>
+#include <winternl.h>
+#include <wrl/client.h>
+
+#include "base/files/file_path.h"
+#include "base/logging.h"
+#include "base/notreached.h"
+#include "base/path_service.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_piece.h"
+#include "base/strings/string_util_win.h"
+#include "base/strings/stringprintf.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/win/atl.h"
+#include "base/win/registry.h"
+#include "base/win/scoped_co_mem.h"
+#include "base/win/scoped_com_initializer.h"
+#include "base/win/win_util.h"
+#include "base/win/windows_version.h"
+#include "chrome/install_static/install_util.h"
+#include "chrome/installer/util/shell_util.h"
+
+// Most of source code in this file comes from firefox's SetDefaultBrowser -
+// https://github.com/mozilla/gecko-dev/blob/master/toolkit/mozapps/defaultagent/SetDefaultBrowser.cpp
+
+namespace {
+
+constexpr wchar_t kUserChoiceKey[] = L"UserChoice";
+constexpr wchar_t kProgIDKey[] = L"ProgID";
+constexpr wchar_t kHashKey[] = L"Hash";
+
+inline DWORD WordSwap(DWORD v) {
+  return (v >> 16) | (v << 16);
+}
+
+std::unique_ptr<DWORD[]> CNGMD5(const unsigned char* bytes, ULONG bytes_len) {
```

Comment:
so `bcrypt.lib` is only used for MD5? we should just use `base/hash/md5.h` and do not pull `bcrypt.lib` for that.

Reply:
Addressed by https://github.com/brave/brave-core/pull/10879/commits/f2560c456fed01b3507e327c6d9d7eded0bd8915
PTAL :)